cluster-autoscaler/core/scale_up.go

```diff
@@ -229,6 +227,19 @@ func getNodeInfoCoresAndMemory(nodeInfo *schedulercache.NodeInfo) (int64, int64)
 	return getNodeCoresAndMemory(nodeInfo.Node())
 }
 
+type skippedReasons struct {
+	message []string
+}
+
+func (sr *skippedReasons) Reasons() []string {
+	return sr.message
+}
+
+var (
+	notReadyReason        = &skippedReasons{[]string{"not ready for scale-up"}}
```

Comment:
We know why the group can be not ready for scale-up, but I feel this doesn't really clarify things for users. Maybe we can come up with a better message? Possibly a few different messages, given that this one is used for a few different things?

Reply:
Agreed we need some catch-all message for "stuff didn't work, sorry" (aka 500). But the 2 most likely reasons to get this would be:
1. group is unhealthy
2. group is in backoff
We can distinguish those and they sound at least somewhat meaningful to user. WDYT?

Reply:
Added new message for the back-off case, PTAL. This requires extra check (duplicated in IsNodeGroupSafeToScaleUp) to separate it from unhealthy-because-of-sizes case.